pkg/newmapper/mapping.go

```diff
@@ -0,0 +1,31 @@
+package newmapper
+
+import (
+	"github.com/leanix/leanix-k8s-connector/pkg/kubernetes"
+	"github.com/leanix/leanix-k8s-connector/pkg/mapper"
+	"k8s.io/client-go/rest"
+)
+
+func ScanWorker(clusterName string, config *rest.Config) ([]mapper.KubernetesObject, error) {
```

Comment:
It sounds like you want to call ScanWorker like a function, but a function should have a verb inside, like scanKubernetes.
You could however make it more functional like we do in the cloud-connector. ScanWorker would then return a function which will return `([]mapper.KubernetesObject, `error)`` and after calling something like `worker := ScanWorker(clusterName, config)` you would then call `worker()`

Reply:
Thanks for the input, what would be the advantage of the second approach? At first glance it just seems like it would add more code?

Reply:
The advantage would only be visible once you would have several workers with different behaviour just as we do in the cloud-connector. Here we can keep it simple.

Reply:
Makes sense, I will take a look in how its implemented on cloud connector side, even if just as ideas for the future :)